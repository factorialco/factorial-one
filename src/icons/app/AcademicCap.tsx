import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAcademicCap = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path
      stroke="currentColor"
      strokeLinejoin="round"
      d="m3.16 8.536 8.654-3.462a.5.5 0 0 1 .372 0l8.653 3.462a.5.5 0 0 1 0 .928l-8.653 3.462a.5.5 0 0 1-.372 0L3.161 9.464a.5.5 0 0 1 0-.928Z"
    />
    <Path
      stroke="currentColor"
      strokeLinejoin="round"
      d="M19 10v5.67a.5.5 0 0 1-.303.46l-6.5 2.786a.5.5 0 0 1-.394 0l-6.5-2.786A.5.5 0 0 1 5 15.67V10"
    />
    <Path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      d="m12 9 4 2v2.5"
    />
  </Svg>
);
const ForwardRef = forwardRef(SvgAcademicCap);
export default ForwardRef;
