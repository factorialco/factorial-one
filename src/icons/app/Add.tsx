import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAdd = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M12 5v14M12 12H5h14"
    />
  </Svg>
);
const ForwardRef = forwardRef(SvgAdd);
export default ForwardRef;
