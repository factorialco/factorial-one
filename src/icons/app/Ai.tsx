import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAi = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path
      stroke="currentColor"
      strokeLinejoin="round"
      d="M4 13c3.314 0 6-3.134 6-7 0 3.866 2.686 7 6 7-3.314 0-6 3.134-6 7 0-3.866-2.686-7-6-7Z"
    />
    <Path
      fill="currentColor"
      d="M17.5 3.35a.65.65 0 0 1 .65.65A1.85 1.85 0 0 0 20 5.85a.65.65 0 1 1 0 1.3A1.85 1.85 0 0 0 18.15 9a.65.65 0 1 1-1.3 0A1.85 1.85 0 0 0 15 7.15a.65.65 0 1 1 0-1.3A1.85 1.85 0 0 0 16.85 4a.65.65 0 0 1 .65-.65"
    />
  </Svg>
);
const ForwardRef = forwardRef(SvgAi);
export default ForwardRef;
