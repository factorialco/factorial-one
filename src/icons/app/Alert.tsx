import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAlert = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path stroke="currentColor" strokeLinecap="round" d="M12 14V7M12 17.1V17" />
  </Svg>
);
const ForwardRef = forwardRef(SvgAlert);
export default ForwardRef;
