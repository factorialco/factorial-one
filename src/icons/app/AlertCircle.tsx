import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAlertCircle = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path
      fill="currentColor"
      d="M12 20a8 8 0 1 0 0-16 8 8 0 0 0 0 16m-.967-11.94a.97.97 0 1 1 1.935 0l-.303 3.9a.667.667 0 0 1-1.33 0zM13 15a1 1 0 1 1-2 0 1 1 0 0 1 2 0"
    />
  </Svg>
);
const ForwardRef = forwardRef(SvgAlertCircle);
export default ForwardRef;
