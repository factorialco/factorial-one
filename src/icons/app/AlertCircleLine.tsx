import Svg, { Circle, Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAlertCircleLine = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Circle cx={12} cy={12} r={8} stroke="currentColor" />
    <Path stroke="currentColor" strokeLinecap="round" d="M12 12V9M12 15.1V15" />
  </Svg>
);
const ForwardRef = forwardRef(SvgAlertCircleLine);
export default ForwardRef;
