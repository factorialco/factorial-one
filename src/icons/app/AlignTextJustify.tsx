import Svg, { Path } from "react-native-svg";
import type { SvgProps } from "react-native-svg";
import { Ref, forwardRef } from "react";
const SvgAlignTextJustify = (props: SvgProps, ref: Ref<Svg>) => (
  <Svg
    fill="none"
    viewBox="0 0 24 24"
    className={props.className}
    ref={ref}
    {...props}
  >
    <Path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M5 8h14M5 12h14M5 16h14"
    />
  </Svg>
);
const ForwardRef = forwardRef(SvgAlignTextJustify);
export default ForwardRef;
