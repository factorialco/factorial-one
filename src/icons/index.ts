import * as AppIcons from "./app";
import * as ModuleIcons from "./modules";
export { AppIcons, ModuleIcons };
export * from "./types";
export { Icon } from "../components/Icon";
