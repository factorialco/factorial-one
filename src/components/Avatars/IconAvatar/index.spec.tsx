import { render } from "@testing-library/react-native";
import React from "react";
import { IconAvatar } from "./";
import { Placeholder } from "../../../icons/app";

describe("IconAvatar", () => {
  it("Snapshot - different sizes", () => {
    const sizes = ["sm", "md", "lg"] as const;

    sizes.forEach((size) => {
      const { toJSON } = render(<IconAvatar icon={Placeholder} size={size} />);

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
