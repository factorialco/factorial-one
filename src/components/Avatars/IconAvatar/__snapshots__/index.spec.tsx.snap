// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAvatar Snapshot - different sizes 1`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary h-6 w-6 rounded-sm"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    className="shrink-0 w-4 h-4 stroke-sm text-f1-foreground-secondary"
    fill="none"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={12}
        cy={12}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
          ]
        }
        r={8}
        stroke={
          {
            "type": 2,
          }
        }
      />
      <RNSVGPath
        d="m17.5 6.5-11 11M17.5 17.5l-11-11"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinejoin={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`IconAvatar Snapshot - different sizes 2`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary h-9 w-9 rounded-md"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    className="shrink-0 w-5 h-5 stroke-md text-f1-foreground-secondary"
    fill="none"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={12}
        cy={12}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
          ]
        }
        r={8}
        stroke={
          {
            "type": 2,
          }
        }
      />
      <RNSVGPath
        d="m17.5 6.5-11 11M17.5 17.5l-11-11"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinejoin={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`IconAvatar Snapshot - different sizes 3`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary h-10 w-10 rounded-lg"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    className="shrink-0 w-6 h-6 stroke-lg text-f1-foreground-secondary"
    fill="none"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGCircle
        cx={12}
        cy={12}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
          ]
        }
        r={8}
        stroke={
          {
            "type": 2,
          }
        }
      />
      <RNSVGPath
        d="m17.5 6.5-11 11M17.5 17.5l-11-11"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinejoin={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
